docs/ref/checks.txt

```diff
@@ -430,9 +430,10 @@ The following checks are run if you use the :option:`check --deploy` option:
   connections, you may want to either set this setting to ``True`` or configure
   a load balancer or reverse-proxy server  to redirect all connections to HTTPS.
 * **security.W009**: Your :setting:`SECRET_KEY` has less than 50 characters or
-  less than 5 unique characters. Please generate a long and random
-  ``SECRET_KEY``, otherwise many of Django's security-critical features will be
-  vulnerable to attack.
+  less than 50 unique characters, or it's a Django-generated secret key
+  starting with Django-insecure. Please generate a long and random
+  :setting:``SECRET_KEY``, otherwise many of Django's security-critical
+  features will be vulnerable to attack.
```

Comment:
With the change to the wording of the message, the issue with "django-generated" should be gone. I've also added backticks around the string here:

```suggestion
* **security.W009**: Your :setting:`SECRET_KEY` has less than 50 characters,
  less than 5 unique characters, or it is prefixed with ``'django-insecure-'``
  indicating that it was generated automatically by Django. Please generate a
  long and random ``SECRET_KEY``, otherwise many of Django's security-critical
  features will be vulnerable to attack.
```

(Again, this suggestion may not apply cleanly in GitHub.)